sql/analyzer/memo.go

```diff
@@ -111,14 +112,30 @@ func (m *Memo) optimizeMemoGroup(grp *exprGroup) error {
 		if err != nil {
 			return err
 		}
+
+		if grp.relProps.distinct.IsHash() {
+			var dCost float64
+			if sortedOnDistinct(n) {
+				n.setDistinct(sortedDistinctOp)
+			} else {
+				n.setDistinct(hashDistinctOp)
+				dCost, err = m.c.EstimateCost(m.ctx, &distinct{child: grp}, m.statsRw)
+				if err != nil {
+					return err
+				}
+			}
+			relCost += dCost
+		} else {
+			n.setDistinct(noDistinctOp)
```

Comment:
Is this a no-op?

Reply:
currently no. i was flip-flopping on whether the 0-code should be "unknown" or "no distinct"